Stack/Opc.Ua.Core/Stack/State/DataTypeState.cs

```diff
@@ -179,21 +180,33 @@ public override void Update(ISystemContext context, BinaryDecoder decoder, Attri
             switch (attributeId)
             {
                 case Attributes.DataTypeDefinition:
-                    {
-                        ExtensionObject dataTypeDefinition = m_dataTypeDefinition;
+                {
+                    ExtensionObject dataTypeDefinition = m_dataTypeDefinition;
 
-                        if (OnReadDataTypeDefinition != null)
-                        {
-                            result = OnReadDataTypeDefinition(context, this, ref dataTypeDefinition);
-                        }
+                    if (OnReadDataTypeDefinition != null)
+                    {
+                        result = OnReadDataTypeDefinition(context, this, ref dataTypeDefinition);
+                    }
 
-                        if (ServiceResult.IsGood(result))
+                    if (ServiceResult.IsGood(result))
+                    {
+                        if (dataTypeDefinition?.Body is StructureDefinition structureType &&
+                            structureType.DefaultEncodingId.IsNullNodeId)
```

Comment:
Is this piece of code covering enum datatypes as well? or  they don't need special handling?

Reply:
good point, EnumDefinition does not have a DefaultEncodingId as a member, I wonder why it is only needed for struct... I believe enum are encoded as int anyway, so it may not even matter